### DISCUSSION TITLE
Help please.     403 otp_expired error for redirected email invitation in production only (redirect works in development)

### DISCUSSION THREAD

**Discussion by SgtErnestBilko** [supabase/supabase#28193] [Questions]
I am using Supabase Auth to send email invitations via service user.  This works well in development.
```
const { error } = await supabase.auth.admin.inviteUserByEmail(
        email,
        {
            redirectTo: redirectTo
        })
```

I develop on a separate lan machine behind a proxy server and in development the 'redirectTo' url is defined as https://xxxx.net/adddetails)

My site URL is https://xxxx.net
My redirect URL is https//xxxx.net/**
My invite email template href is "{{ .SiteURL }}/confirm?token_hash={{ .TokenHash }}&type=email&next={{ .RedirectTo }}"

This works fine in development.  I can send email, click link and be redirected to the adddetails page which allows addition of name and password to profiles and auth respectively.

In production I change:
redirectTo: https://abcd.org/adddetails
site URL is https://abcd.org
redirect URL is https//abcd.org/**
email href is unchanged

When I click the email link the user is created in Supabase but the user is not redirected to the adddetails page.

This is is my (auth)/confirm/route.tsx code

```
const defaultUrl = new URL("https://abcd.org")

const redirectPath = '/adddetails'

export async function GET(request: NextRequest) {
  const { searchParams } = new URL(request.url)
  const token_hash = searchParams.get('token_hash')
  const type = searchParams.get('type') as EmailOtpType | null
  const next = searchParams.get('next') ?? '/'

  const requestURL = new URL(next)
  const origin = requestURL.origin
  const path = requestURL.pathname.trim()
  let redirectTo = new NextURL(origin)


  //Security test if the full url sent isn't the same (or isn't complete) as that received then redirect to /
  if (redirectTo.origin != defaultUrl.origin || redirectPath != path) {
    console.error('failed security test: url ', redirectTo.origin, ' should = ', defaultUrl.origin, ' and path ', redirectPath, ' should equal ', path)
    redirectTo.pathname = '/'
    return NextResponse.redirect(redirectTo)
  }

  //  Verify user and send to updateDetails in adddetails so user can add password and name
  if (token_hash && type) {
    const supabase = createClient()

    const { error } = await supabase.auth.verifyOtp({
      type,
      token_hash,
    })
    if (!error) {
      redirectTo.pathname = path
      return NextResponse.redirect(redirectTo)
    }
    if (error) {
      console.error('verify failed in confirm/route.tsx: ' + error)
    }
  }

  // return the user to / if there was an error verifying
  redirectTo.pathname = '/'
  return NextResponse.redirect(redirectTo)
}
```

In production a redacted example of the email link is : https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fabcd.org%2Fconfirm%3Ftoken_hash%3---blahblah--26type%3Demail%26next%3Dhttps%253a%252f%252fabcd.org%252fadddetails&data=05%7C02%7C%7C8b4fb2246cb148b17e9708dcacaf2ad3%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C6385xxx278634%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2lxxxxxCI6Mn0%3D%7C0%7C%7C%7C&sdata=kC7joYf%2BExX%2F4xxxxx7bw%2F3047AdgRchbQ4NNR7E%3D&reserved=0

In production supabase.auth.verifyOtp produces an error (below) and the user is redirected to '/'

```
eG [AuthApiError]: Email link is invalid or has expired
    at e5 (/var/task/.next/server/chunks/293.js:1:110288)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async ae (/var/task/.next/server/chunks/293.js:1:111287)
    at async e7 (/var/task/.next/server/chunks/293.js:1:110943)
    at async a_.verifyOtp (/var/task/.next/server/chunks/293.js:1:128467)
    at async h (/var/task/.next/server/app/(auth)/confirm/route.js:1:1983)
    at async /var/task/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:36258
    at async eR.execute (/var/task/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:26874)
    at async eR.handle (/var/task/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:37512)
    at async es (/var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js:16:25465) {
  __isAuthError: true,
  status: 403,
  code: 'otp_expired'

```

Email OTP Expiration is set to: 3599 seconds
I have enabled a custom SMTP in Settings > Configuration > Authentication 

Is this a bug ?

Any help would be much appreciated.

Ian

**Comment by SgtErnestBilko** [comment]
Does anyone have an idea???

**Comment by SgtErnestBilko** [comment]
Sorry to bump this but I'm stuck

**Comment by SgtErnestBilko** [comment]
I wonder ....... is the answer so simple it doesn't deserve help..... or so difficult no one can help me ?

**Comment by softmarshmallow** [comment]
Having a similar issue with magic link signup.

```
error: eU [AuthApiError]: Email link is invalid or has expired
```

The api started throwing this message ever since a day or two ago. no code has changed. hmm..

**Comment by silentworks** [comment]
The issue here is potentially email prefetching happening by your email client. https://supabase.com/docs/guides/auth/auth-email-templates#email-prefetching

**Reply by SgtErnestBilko** [reply]
Thank you I appreciate your reply.   Sounds like it could be it.   I will check in out in the next few days .
